### DISCUSSION TITLE
New data source - required changes?

### DISCUSSION THREAD

**Discussion by danielkoohmarey-bc** [foxglove/community#24] [Q&A]
Hello, I'm investigating the amount of work associated with supporting a custom data source / data format that contains similar payloads/fields as existing ros messages. Do you have any pointers with the relevant files that would need to be changed, or tutorial on supporting a new data source? Thanks!

**Comment by jtbandes** [comment]
Hi, can you say more about your data source? Is it a "live" connection or pre-recorded data? For live data, see: https://foxglove.dev/docs/connection/foxglove-websocket

For recording data, we are also working on a new file format at https://github.com/foxglove/mcap

We also plan to add more documentation around bringing custom data into Foxglove Studio.

**Reply by danielkoohmarey-bc** [reply]
Both use cases (live&recorded) would be useful, the format of the data in both cases would appear to be the same though. It is somewhat similar to your mcap format (as its a custom serialization protocol). Let me know once have have mcap support integrated if you can point me at the relevant files you modified to support it, or when this new documentation is completed. Thanks!

**Reply by esthersweon** [reply]
Hey @danielkoohmarey-bc - please let me know if the existing documentation for [loading custom data in Foxglove Studio](https://foxglove.dev/docs/studio/connection/custom) answers all your questions, or if there are any gaps that you see missing.
